### DISCUSSION TITLE
Setup timeout for the request

### DISCUSSION THREAD

**Discussion by yjl9903** [unocss/unocss#1345] [General]
There is some network issues in my env. It fails to request google fonts. Vite stuck in infinite waiting, and has no output. Can we setup a timeout for this http request to avoid long time waiting?

https://github.com/unocss/unocss/blob/d290c847cb261bc50d31773a6841cee09df8c8f6/packages/preset-web-fonts/src/index.ts#L47-L56

**Comment by antfu** [comment]
Sure, ohmyfetch also supports `retry` and `timeout` if IIRC. PR welcome

**Reply by yjl9903** [reply]
I only see it supports [retry](https://github.com/unjs/ohmyfetch#%EF%B8%8F-auto-retry), but no timeout
